rs/ledger_suite/tests/sm-tests/src/lib.rs

```diff
@@ -2823,6 +2823,32 @@ pub fn icrc1_test_upgrade_serialization_fixed_tx<T>(
     }
 }
 
+pub fn test_downgrade_from_incompatible_version<T>(
+    ledger_wasm_nextledgerversion: Vec<u8>,
+    ledger_wasm: Vec<u8>,
+    encode_init_args: fn(InitArgs) -> T,
+) where
+    T: CandidType,
+{
+    // Setup ledger with unsupported future version.
+    let (env, canister_id) = setup(ledger_wasm_nextledgerversion, encode_init_args, vec![]);
+
+    match env.upgrade_canister(
+        canister_id,
+        ledger_wasm,
+        Encode!(&LedgerArgument::Upgrade(None)).unwrap(),
+    ) {
+        Ok(_) => {
+            panic!("Upgrade from future ledger version should fail!")
+        }
+        Err(e) => {
+            assert!(e
+                .description()
+                .contains("Trying to downgrade from incompatible version"))
+        }
+    };
+}
+
```

Comment:
For completeness, would it also make sense to add:

- Upgrading to the same wasm always works
- Downgrading from `ledger_wasm_nextledgerversion` to current mainnet version succeeds (although it shouldn't) - it's a two-version downgrade which we anyway wouldn't support, but since the current mainnet version doesn't have the field and cannot check, the downgrade will actually succeed

Reply:
Done!